frappe/desk/doctype/number_card/number_card.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe.model.document import Document
+
+class NumberCard(Document):
+	pass
+
+
+def get_permission_query_conditions(user):
+
+	if not user:
+		user = frappe.session.user
+
+	if user == 'Administrator':
+		return
+
+	roles = frappe.get_roles(user)
+	if "System Manager" in roles:
+		return None
+
+	allowed_doctypes = tuple(frappe.permissions.get_doctypes_with_read())
+
+	return '''
+			`tabNumber Card`.`document_type` in {allowed_doctypes}
+		'''.format(
+			allowed_doctypes=allowed_doctypes,
+		)
+
+def has_permission(doc, ptype, user):
+	roles = frappe.get_roles(user)
+	if "System Manager" in roles:
+		return True
+
+	allowed_doctypes = tuple(frappe.permissions.get_doctypes_with_read())
+	if doc.document_type in allowed_doctypes:
+		return True
+
+	return False
+
+@frappe.whitelist()
+def get_result(doc, to_date=None):
+	doc = frappe.parse_json(doc)
+	fields = []
+	sql_function_map = {
+		'Count': 'count',
+		'Sum': 'sum',
+		'Average': 'avg',
+		'Minimum': 'min',
+		'Maximum': 'max'
+	}
+
+	function = sql_function_map[doc.function]
+
+	if function == 'count':
+		fields = ['{function}(*) as result'.format(function=function)]
+	else:
+		fields = ['{function}({based_on}) as result'.format(function=function, based_on=doc.aggregate_function_based_on)]
+
+	filters = frappe.parse_json(doc.filters_json)
+
+	if to_date:
+		filters.append([doc.document_type, 'creation', '<', to_date, False])
+
+	number = frappe.db.get_all(doc.document_type, fields = fields, filters = filters)[0]['result']
```

Comment:
You can get a `IndexError` or `KeyError` here if the list is empty or `result` key is not presetn